Initial port of nuget package docs (#20)
microsoft/winforms-designer-extensibility: README.md

## Documents

> * [Using Visual Studio Solution Templates for creating Control Designers/Type Editors for the Out-Of-Process .NET Core WinForms Designer](https://github.com/microsoft/winforms-designer-extensibility/tree/main/Templates/TypeEditor).

Reviewer: ```suggestion
* [Using Visual Studio solution templates for creating Control Designers/Type Editors for the out-of-process .NET Windows Forms designer](https://github.com/microsoft/winforms-designer-extensibility/tree/main/Templates/TypeEditor).
```

> * [Structuring Control Library nuget packages](/docs/sdk/control-library-nuget-package-spec.md)

Reviewer: ```suggestion
* [Structuring Control Library NuGet packages](/docs/sdk/control-library-nuget-package-spec.md)
```

> * [Using Visual Studio Solution Templates for creating Control Designers/Type Editors for the Out-Of-Process .NET Core WinForms Designer](https://github.com/microsoft/winforms-designer-extensibility/tree/main/Templates/TypeEditor).
> * [Structuring Control Library nuget packages](/docs/sdk/control-library-nuget-package-spec.md)

Reviewer: I'd reword these to be more grammatically correct. E.g., to something like:
```suggestion
* [How to create Control Designers/Type Editors for the out-of-process .NET Windows Forms designer using Visual Studio solution templates](https://github.com/microsoft/winforms-designer-extensibility/tree/main/Templates/TypeEditor).
* [How to structure Control Library NuGet packages](/docs/sdk/control-library-nuget-package-spec.md)
```

Are all these docs specific to the **out-of-process** or **out-of-process .NET** designer? If so, then add this to the `## Documents` header and remove from the link text.
